src/libsystemd/sd-resolve/sd-resolve.c

```diff
@@ -778,12 +764,10 @@ static int handle_response(sd_resolve *resolve, const Packet *packet, size_t len
                 size_t l;
                 struct addrinfo *prev = NULL;
 
-                assert(length >= sizeof(AddrInfoResponse));
-                assert(q->type == REQUEST_ADDRINFO);
+                assert_return(length >= sizeof(NameInfoResponse), -EBADMSG);
+                assert_return(q->type == REQUEST_NAMEINFO, -EBADMSG);
```

Comment:
`REQUEST_ADDRINFO` instead of `REQUEST_NAMEINFO`.

Reply:
Yikes.